source/speech/__init__.py

```diff
@@ -1244,31 +1328,51 @@ def getControlFieldSpeech(attrs,ancestorAttrs,fieldType,formatConfig=None,extraD
 
 	# Determine what text to speak.
 	# Special cases
-	if childControlCount and fieldType=="start_addedToControlFieldStack" and role==controlTypes.ROLE_LIST and controlTypes.STATE_READONLY in states:
+	if(
+			childControlCount
+			and fieldType=="start_addedToControlFieldStack"
+			and role==controlTypes.ROLE_LIST
+			and controlTypes.STATE_READONLY in states
+	):
 		# List.
-		# #7652: containerContainsText variable is set here, but the actual generation of all other output is handled further down in the general cases section.
+		# #7652: containerContainsText variable is set here, but the actual generation of all other output is
+		# handled further down in the general cases section.
 		# This ensures that properties such as name, states and level etc still get reported appropriately.
 		# Translators: Number of items in a list (example output: list with 5 items).
 		containerContainsText=_("with %s items")%childControlCount
 	elif fieldType=="start_addedToControlFieldStack" and role==controlTypes.ROLE_TABLE and tableID:
 		# Table.
 		rowCount=(attrs.get("table-rowcount-presentational") or attrs.get("table-rowcount"))
 		columnCount=(attrs.get("table-columncount-presentational") or attrs.get("table-columncount"))
-		return " ".join((
-			nameText,
-			roleText,
-			stateText, 
-			getSpeechTextForProperties(
+		tableSeq = nameSequence

```

Comment:
Note that this also touches the nameSequence when extending. Also for uniformity, I'd change its name.

```suggestion
		tableSequence = nameSequence[:]
```